### PR TITLE
Avoid ipv4 static and DHCP IPaddress coexistence

### PR DESCRIPTION
**Pull request by raviteja-b**
Currently IPv4 Static and DHCP IP addresses coexists, but Static IP configuration
can't work while DHCP enabled on the interface.

This commits avoids coexistence, disables DHCPv4 while configuring static IPv4 address 
and deletes static address while enabling DHCPv4.

This is downstream only change for now, needs discussions in the upstream community

Tested by:
Enable DHCPv4 with multiple static IPv4 addresses configured on interface.
Configure Static addresses when DHCPv4 enabled on interface.
Verified DHCPv6 Enable/Disable test cases.
Verified Static IPv6 address  configuration while enabling DHCP

### CODE REVIEW

File: src/ethernet_interface.cpp [hunk 1]
```diff
@@ -468,8 +514,37 @@ bool EthernetInterface::dhcp6(bool value)
...
+void EthernetInterface::deleteStaticIPv4Addresses()
+{
+    std::unique_ptr<IPAddress> ptr;
+    for (auto it = addrs.begin(); it != addrs.end();)
+    {
+        if ((it->second->origin() == IP::AddressOrigin::Static) &&
+            (it->second->type() == IP::Protocol::IPv4))
+        {
+            ptr = std::move(it->second);
+            it = addrs.erase(it);
```
Comment (asmithakarun) [inline src/ethernet_interface.cpp:526]: Hope this means that the ip address object also will be deleted.

**Review by asmithakarun — APPROVED** [review]

File: src/ethernet_interface.cpp (hunk 1 above)
Comment (raviteja-b) [reply src/ethernet_interface.cpp:526]: yes

File: src/ethernet_interface.cpp [hunk 2]
```diff
@@ -468,20 +514,44 @@ bool EthernetInterface::dhcp6(bool value)
...
+    if ((value == DHCPConf::v4) || (value == DHCPConf::both))
+    {
+        // Delete all IPv4 static addresses while enabling DHCP
+        deleteStaticIPv4Addresses();
+    }
+
     auto new4 = EthernetInterfaceIntf::dhcp4(value == DHCPConf::v4 ||
                                              value == DHCPConf::both);
-    auto old6 = EthernetInterfaceIntf::dhcp6();
     auto new6 = EthernetInterfaceIntf::dhcp6(value == DHCPConf::v6 ||
```
Comment (sunharis) [inline src/ethernet_interface.cpp:550]: What are these new4 and new6 are used for here? This is common code for IPv6 as well. So i think better to cover IPv6 side of DHCP things as well with this patch

**Comment by geissonator** [comment]
CI compile failure:
```
../src/ethernet_interface.cpp:548:10: error: unused variable ‘new4’ [-Werror=unused-variable]
  548 |     auto new4 = EthernetInterfaceIntf::dhcp4(value == DHCPConf::v4 ||
      |          ^~~~

**Comment by asmithakarun** [comment]
Seems CI is failing for clang formatting issues. can you fix it? 
@raviteja-b

**Comment by raviteja-b** [comment]
seem we need https://github.com/ibm-openbmc/phosphor-networkd/pull/82 to pass CI

**Review by sunharis — APPROVED** [review]
Please test & add IPv6 test results as well

**Comment by geissonator** [comment]
> will restart CI after above PR gets merged

Merged, may need to rebase this PR to pick it up

**Comment by raviteja-b** [comment]
Create new PR https://github.com/ibm-openbmc/phosphor-networkd/pull/83

**Pull request closed without merging**